microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/HelidonTelemetryContainerFilter.java

```diff
@@ -278,19 +278,20 @@ private String resolveTarget(ContainerRequestContext requestContext) {
     private void handleBaggage(ContainerRequestContext containerRequestContext, Context context) {
         List<String> baggageProperties = containerRequestContext.getHeaders().get("baggage");
         if (baggageProperties != null) {
+            var baggageBuilder = Baggage.builder();
             for (String b : baggageProperties) {
                 String[] split = b.split("=");
                 if (split.length == 2) {
                     String[] valueAndMetadata = split[1].split(";");
                     String value = valueAndMetadata.length > 0 ? valueAndMetadata[0] : "";
                     String metadata = valueAndMetadata.length > 1 ? valueAndMetadata[1] : "";
-                    Baggage.builder()
-                            .put(split[0], value, BaggageEntryMetadata.create(metadata))
-                            .build()
-                            .storeInContext(context)
-                            .makeCurrent();
+                    baggageBuilder
+                            .put(split[0], value, BaggageEntryMetadata.create(metadata));
```

Comment:
Minor nit and I know it's not in the scope of this PR: the `split` calls will recompile their regex patterns each time